lib/full-site-editing/template-loader.php

```diff
@@ -69,6 +70,19 @@ function gutenberg_override_query_template( $template, $type, array $templates =
 	$current_template_slug       = basename( $template, '.php' );
 	$current_block_template_slug = is_object( $current_template ) ? $current_template->slug : false;
 	foreach ( $templates as $template_item ) {
+
+		// if the theme is a child theme we want to check if a php template exists
+		// and that a corresponding block template from the theme and not the parent doesn't exist
+		$has_php_template   = file_exists( get_stylesheet_directory() . '/' . $type . '.php' );
+		$has_block_template = false;
+		$block_template     = _gutenberg_get_template_file( 'wp_template', $type );
+		if ( $block_template !== null && $block_template['theme'] == wp_get_theme()->get_stylesheet() ) {
+			$has_block_template = true;
```

Comment:
I wonder if lines 77-81 would be better encapsulated in a separate function?

Reply:
I thought about but I didn't do it because I thought it was very unlikely that such a function would be reused since it would be very specific. And the rest of the gutenberg_override_query_template function is using similar checks so I thought this way was a bit more consistent. I'm happy to do it but I'm not sure what we gain from doing so.